internal/ignition/ignition.go

```diff
@@ -1264,11 +1273,7 @@ func MergeIgnitionConfig(base []byte, overrides []byte) (string, error) {
 
 	// Validate after we marshal to use the Parse functions
 	var report report.Report
-	if baseConfig.Ignition.Version == "3.1.0" {
-		_, report, err = config_31.Parse(res)
-	} else {
-		_, report, err = config_latest.Parse(res)
-	}
+	_, report, err = config_latest.ParseCompatibleVersion(res)
```

Comment:
Does this change the behavior? What if someone put 3.2 features in a 3.1 version override? Would this catch that case?

Reply:
Outdated, keeping the old behaviour (i.e. separately parsing 3.1.0 and 3.2.0)